src/app/public/src/modules/page-anchor/page-anchor.component.ts

```diff
@@ -1,109 +1,106 @@
 import {
-  ChangeDetectorRef,
   Component,
   ElementRef,
-  OnInit,
   AfterViewInit,
-  Input
+  Input,
+  ChangeDetectorRef,
+  OnDestroy
 } from '@angular/core';
 import {
-  StacheWindowRef,
-  StacheRouteService
+  StacheWindowRef, StacheRouteService
 } from '../shared';
 import { StacheNavLink } from '../nav';
 import { StachePageAnchorService } from './page-anchor.service';
+import { BehaviorSubject } from 'rxjs';
 
 @Component({
   selector: 'stache-page-anchor',
   templateUrl: './page-anchor.component.html',
   styleUrls: ['./page-anchor.component.scss']
 })
-export class StachePageAnchorComponent implements OnInit, StacheNavLink, AfterViewInit {
-
-  public name: string;
+export class StachePageAnchorComponent implements StacheNavLink, AfterViewInit, OnDestroy {
+  public name: string = '';
   public fragment: string;
   public path: string[];
-  public order: number;
   public offsetTop: number;
+  public anchor = new BehaviorSubject<StacheNavLink>({
+    name: this.name,
+    fragment: this.fragment,
+    path: this.path,
+    offsetTop: this.offsetTop
+  });
 
   @Input()
   public anchorId?: string;
 
   public constructor(
-    private routerService: StacheRouteService,
     private elementRef: ElementRef,
     private windowRef: StacheWindowRef,
-    private cdRef: ChangeDetectorRef,
-    private anchorService: StachePageAnchorService) {
-    this.name = '';
-  }
-
-  public ngOnInit(): void {
-    this.name = this.getName();
-    this.fragment = this.getFragment();
-    this.path = [this.routerService.getActiveUrl()];
-  }
+    private anchorService: StachePageAnchorService,
+    private cdr: ChangeDetectorRef,
+    private routeService: StacheRouteService) {
+      this.anchorService.bodyHeightChanged.subscribe(() => {
+        this.updatePageAnchor();
+      });
+    }
 
   public scrollToAnchor(): void {
     this.windowRef.nativeWindow.document.querySelector(`#${this.fragment}`).scrollIntoView();
   }
 
   public ngAfterViewInit(): void {
-    this.name = this.getName();
-    this.fragment = this.getFragment();
-    this.offsetTop = this.getOffsetTop();
-    this.getOrder();
-    this.registerAnchor();
-    this.cdRef.detectChanges();
+    this.generateAnchor(this.elementRef.nativeElement, this.anchorId);
+    this.cdr.detectChanges();
   }
 
-  private getName(): string {
-    return this.elementRef.nativeElement.textContent.trim();
+  public ngOnDestroy() {
+    this.anchor.complete();
   }
 
-  private getFragment(): string {
-    return this.anchorId || this.name
-      .toLowerCase()
-      .replace(/ /g, '-')
-      .replace(/[^\w-]+/g, '');
-  }
+  public generateAnchor(element: any, anchorId: any = undefined) {
+    this.name = this.getName(element);
+    this.fragment = this.getFragment(this.name, anchorId);
```

Comment:
We should reformat this to.
```
this.fragment = this.anchorId || this.getFragment(this.name);
```
right now there is no need to be passing that anchorId around.